tests/libvmi/expecter.go

```diff
@@ -0,0 +1,406 @@
+	panicOnError(err)
+	expecter, _, err := NewConsoleExpecter(virtClient, vmi, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	b := append([]expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: "localhost login:"},
+		&expect.BSnd{S: "root\n"},
+		&expect.BExp{R: "localhost:~\\#"}})
+	res, err := expecter.ExpectBatch(b, 180*time.Second)
+	if err != nil {
+		log.DefaultLogger().Object(vmi).Infof("Login: %v", res)
+		expecter.Close()
+		return nil, err
+	}
+
+	err = configureConsole(expecter, "localhost:~\\#", false)
+	if err != nil {
+		expecter.Close()
+		return nil, err
+	}
+	return expecter, err
+}
+
+// LoggedInFedoraExpecter return prepared and ready to use console expecter for
+// Fedora test VM
+func LoggedInFedoraExpecter(vmi *v1.VirtualMachineInstance) (expect.Expecter, error) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	panicOnError(err)
+	expecter, _, err := NewConsoleExpecter(virtClient, vmi, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	b := append([]expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BSnd{S: "\n"},
+		&expect.BCas{C: []expect.Caser{
+			&expect.Case{
+				// In case the VM's did not get hostname form DHCP server try the default hostname
+				R:  regexp.MustCompile(`localhost login: `),
+				S:  "fedora\n",
+				T:  expect.Next(),
+				Rt: 10,
+			},
+			&expect.Case{
+				// Using only "login: " would match things like "Last failed login: Tue Jun  9 22:25:30 UTC 2020 on ttyS0"
+				R:  regexp.MustCompile(vmi.Name + ` login: `),
+				S:  "fedora\n",
+				T:  expect.Next(),
+				Rt: 10,
+			},
+			&expect.Case{
+				R:  regexp.MustCompile(`Password:`),
+				S:  "fedora\n",
+				T:  expect.Next(),
+				Rt: 10,
+			},
+			&expect.Case{
+				R:  regexp.MustCompile(`Login incorrect`),
+				T:  expect.LogContinue("Failed to log in", expect.NewStatus(codes.PermissionDenied, "login failed")),
+				Rt: 10,
+			},
+			&expect.Case{
+				R: regexp.MustCompile(`\$ `),
+				T: expect.OK(),
+			},
+		}},
+		&expect.BSnd{S: "sudo su\n"},
+		&expect.BExp{R: "\\#"},
+	})
+	res, err := expecter.ExpectBatch(b, 3*time.Minute)
+	if err != nil {
+		log.DefaultLogger().Object(vmi).Infof("Login: %+v", res)
+		expecter.Close()
+		return expecter, err
+	}
+
+	err = configureConsole(expecter, "\\#", false)
+	if err != nil {
+		expecter.Close()
+		return nil, err
+	}
+
+	return expecter, configureIPv6OnVMI(vmi, expecter, virtClient, "\\#")
+}
+
+// ReLoggedInFedoraExpecter return prepared and ready to use console expecter for
+// Fedora test VM, when you are reconnecting (no login needed)
+func ReLoggedInFedoraExpecter(vmi *v1.VirtualMachineInstance, timeout int) (expect.Expecter, error) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	panicOnError(err)
+	expecter, _, err := NewConsoleExpecter(virtClient, vmi, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	b := append([]expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: "#"}})
```

Comment:
`\\#` or backtick with `\#`

Reply:
Thanks. This is just a backup in case my laptop blows up, not need to review it, it is bunch of drafts